### DISCUSSION TITLE
Selective syncing of a git repository subfolders

### DISCUSSION THREAD

**Discussion by xplosionmind** [community/community#23655] [New to GitHub]
<p>Hello everybody, I apologize in advance if this sounds as a silly question. I looked for an answer everywhere but I can’t find it: so if you don’t have time to reply, a link to an article will be welcome and very useful anyway.</p>
<p>Let’s get to the point: I’m using GitHub to store the source code of my website. It’s getting quite big, and I’d even like to add a gallery of images, which will make it even bigger. I fear on the long run it will become too large for my computer, and having the whole repository on my computer may become a problem of storage. Is there a way to “selectively” sync which folder to keep locally, while keeping the “heavy” subfolders of the repository (namely the ones containing images and large files) on the remote origin (only on GitHub). Is it possible to do so? Is it absurd?</p>
<p>Thanks a lot!</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Hello <a class="mention" href="/xplosionmind">@xplosionmind</a> and welcome to the community.</p>
<p>I don’t think this is a silly question at all :grinning: On the other hand, I don’t think that this is exclusively a git or GitHub question …</p>
<p>git, as a tool, is specifically designed to handle versioning text files. Out of necessity, it will version binary files such as images also but isn’t optimized for those types of files. So, in your hypothetical situation, not only will you run into a problem of storage on your local computer but you’ll increasingly run into a problem where git doesn’t handle well the types of files you’re asking it to deal with.</p>
<p>Because of this, I would store the large files someplace else such as <a href="https://aws.amazon.com/s3/">Amazon’s S3</a> or <a href="https://azure.microsoft.com/en-in/services/storage/blobs/">Azure Blob Storage</a>. Then, I would have your website make those large files directly available from whichever cloud storage location you select.</p>
<p>It will definitely make things a little more complex to keep track of. But on the other hand, you’ll be storing the differing file types in systems that were optimized for those different requirements.</p>
<p>Let us know if you have more questions.</p>

**Comment by xplosionmind** [comment]
<p>Thanks a lot, you’ve been very helpful!</p>
